### DISCUSSION TITLE
Using Orchard core Open ID as server and as client

### DISCUSSION THREAD

**Discussion by vitalybrandes** [OrchardCMS/OrchardCore#7023] [Q&A]
Hi
I am building complex project for Industrial.
Part of the project based on orchard core.
My plan is to run server with orchard core in **cloud**, and container with orchard core as **local** service. (On PC)
The idea to use OPEN ID in both places.
The "**Client**" connect "**Server**" using Open ID and create user in **Local , "Client environment"**.
3TH side application  (Wpf, core, js whatever) will authenticate with the "**Local**" Orchard core for some data via Graphql or API.

Can i do that? If yes, any HELP with how to use Open ID client in orchard.
Any other solution, something that maybe i miss?

@kevinchalet  I am sure you can say something about? ;-) , please....

**Comment by MichaelPetrinolis** [comment]
From my point of view, you could enable OpenID server feature on 'server' and create an application for each one of the local instances. On each local instance you should enable OpenID server feature and OpenID client feature, configure client feature to authenticate against 'server' and users feature to use the external users only. Define a map between the roles of 'server' and 'client'. Then you can request data from 'local' after authenticating, by using an application in OpenID connect and an appropriate flow for each one of your clients.

You could check the [Use Azure AD as an external identity provider guide](https://docs.orchardcore.net/en/latest/docs/guides/azuread-integration/) how to configure an External Provider and map roles to local users

**Reply by vitalybrandes** [reply]
Thank you @MichaelPetrinolis !
Is there any guide how to configure Client settings for Open Id?

**Reply by MichaelPetrinolis** [reply]
I don't think that there is a guide, but if you follow the [documentation ](https://docs.orchardcore.net/en/latest/docs/reference/modules/OpenId/#openid-configuration) you will be able to configure the client. There are also a ton of examples in blog posts etc. If you want, you can make a PR with a full Orchard Core guide for server and client ;)
tip: request the email also in scopes, apart from openid and profile

**Reply by yassinsayed** [reply]
@vitalybrandes hi did you configure OC client openid

**Comment by hyzx86** [comment]
Hi @vitalybrandes ,  You can refer to this repository

https://github.com/OrchardSkills/OrchardSkills.OrchardCore.OIDC.Vue.js
